src/components/date-facet.tsx

```diff
@@ -0,0 +1,58 @@
+import {DateFacet as HeadlessDateFacet, DateFacetValue} from '@coveo/headless';
+import {FunctionComponent, useEffect, useState} from 'react';
+import dayjs from 'dayjs';
+
+interface DateFacetProps {
+  controller: HeadlessDateFacet;
+  title: string;
+}
+
+export const DateFacet: FunctionComponent<DateFacetProps> = (props) => {
+  const {controller} = props;
+  const [state, setState] = useState(controller.state);
+
+  useEffect(() => controller.subscribe(() => setState(controller.state)), []);
+
+  const parseDate = (date: string) => {
+    return dayjs(date, 'YYYY/MM/DD@HH:mm:ss');
+  };
+
+  const format = (dateStr: string) => {
+    return parseDate(dateStr).format('MMMM D YYYY');
+  };
+
+  const getKeyForRange = (value: DateFacetValue) => {
+    return `[${value.start}..${value.end}${value.endInclusive ? ']' : '['}`;
+  };
+
+  if (!state.values.length) {
+    return (
+      <div className="date-facet">
+        <h3>{props.title}</h3>
+        <div>No facet values</div>
+      </div>
+    );
+  }
+
+  return (
+    <div className="date-facet">
+      <h3>{props.title}</h3>
+      <ul>
+        {state.values.map((value) => (
+          <li key={getKeyForRange(value)}>
+            <input
+              type="checkbox"
+              checked={controller.isValueSelected(value)}
+              onChange={() => controller.toggleSelect(value)}
+              disabled={state.isLoading}
+            />
+            {format(value.start)} to {format(value.end)}{' '}
+            {value.endInclusive ? 'inclusively' : 'exclusively'} (
+            {value.numberOfResults}{' '}
+            {value.numberOfResults === 1 ? 'result' : 'results'})
```

Comment:
Since all of the formatting here depends on the value, we could make a function that takes a value and returns the correct label for the checkbox. Not a big difference, but since it won't be as far indented the linting might come out a bit cleaner